### DISCUSSION TITLE
The troublesome IOS and Mac OS builds

### DISCUSSION THREAD

**Discussion by RechieKho** [iree-gd/iree.gd#57] [General]
Hi, I'm the creator and maintainer of `iree.gd`. So you might be wondering why it doesn't have IOS, Mac OS builds include in the release currently. So here is some in depth details on it.

Cross-compiling to [IOS](https://iree.dev/building-from-source/ios/#build-the-iree-compiler-for-the-host) or other Mac OS of different chip from Mac OS is technically possible, but it is quite troublesome and tiring. Cross-compiling requires the build to compile a full version of IREE host tools prior. If you are cross-compiling from Linux, then you could just download the most recent version of IREE host tools (`iree-dist`) from [the most recent release of IREE](https://github.com/openxla/iree/releases). But in this case, we must build the host tools ourselves, and this is a *very* time consuming process. You need to compile llvm project, and that is an absolute menace and time killer. Hopefully, someone can take on this job of compiling IREE host tools for different platform to ease cross-compilations.

**Comment by RechieKho** [comment]
I proposed having precompiled IREE host tools for Apple products to `IREE`. [Let's see how it goes](https://github.com/openxla/iree/issues/16146)

**Comment by gmmyung** [comment]
Well actually, this issue is pretty much solved by https://github.com/openxla/iree/issues/16109, https://github.com/openxla/iree/pull/16111.
Now IREE runtime build does not require the IREE compiler tools, which means that IREE host tool compilation doesn't take that much time anymore! It takes ~1min to build both the host tools and runtime on my M1 MacBook.

**Reply by RechieKho** [reply]
Cool, thanks for notifying! Will try it out.

**Reply by RechieKho** [reply]
Thanks, the apple builds successfully compile the library. Now I just need to codesign them... (Good thing is that fire is going to help me to do all the hard stuff)

**Reply by fire** [reply]
! Thanks

**Comment by fire** [comment]
I believe this issue is complete.
